src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -609,11 +609,30 @@ void FancyZones::OnDisplayChange(DisplayChangeType changeType) noexcept
             m_currentVirtualDesktopId = currentVirtualDesktopId;
             wil::unique_cotaskmem_string id;
             if (changeType == DisplayChangeType::Initialization &&
-                SUCCEEDED_LOG(StringFromCLSID(m_currentVirtualDesktopId, &id)))
+                SUCCEEDED(StringFromCLSID(m_currentVirtualDesktopId, &id)))
             {
                 JSONHelpers::FancyZonesDataInstance().UpdatePrimaryDesktopData(id.get());
             }
         }
+        if (changeType == DisplayChangeType::Initialization)
+        {
+            // Get currenty active desktops and clean data from persisted storage for all closed desktops.
+            std::vector<GUID> ids{};
+            if (VirtualDesktopUtils::GetVirtualDekstopIds(ids))

```

Comment:
Can we add a new method to the virtual desktop utils class to return the vector of ids?

Reply:
You mean vector if ids, but string type, not GUID structure?

Reply:
Added overloaded function which takes vector of wide strings.